frontend/src/components/RepositorySelectionPanel.vue

```diff
@@ -36,81 +36,100 @@
 </template>
 
 <script lang="ts">
+export default { name: "RepositorySelectionPanel" };
+</script>
+
+<script setup lang="ts">
 import { useStore } from "vuex";
-import { reactive, computed, PropType, watchEffect } from "vue";
-import { ExternalRepositoryInfo, ProjectRepositoryConfig } from "../types";
+import { reactive, computed, onMounted } from "vue";
+import {
+  ExternalRepositoryInfo,
+  OAuthToken,
+  ProjectRepositoryConfig,
+} from "../types";
 
 interface LocalState {
   repositoryList: ExternalRepositoryInfo[];
+  lastAccessToken: string;
   searchText: string;
 }
 
-export default {
-  name: "RepositorySelectionPanel",
-  props: {
-    config: {
-      required: true,
-      type: Object as PropType<ProjectRepositoryConfig>,
-    },
-  },
-  emits: ["next"],
-  setup(props, { emit }) {
-    const store = useStore();
-    const state = reactive<LocalState>({
-      repositoryList: [],
-      searchText: "",
-    });
+const props = defineProps<{ config: ProjectRepositoryConfig }>();
+
+const emit = defineEmits<{
+  (event: "next"): void;
+  (event: "set-token", payload: OAuthToken): void;
+  (event: "set-repository", payload: ExternalRepositoryInfo): void;
+}>();
+
+const store = useStore();
+const state = reactive<LocalState>({
+  repositoryList: [],
+  lastAccessToken: "none",
```

Comment:
convention is to use empty string.

please avoid inventing new mean (bob uses "none", alice uses "empty" and it will be a mess)